zettarepl/snapshot/destroy.py

```diff
@@ -16,6 +17,17 @@ def destroy_snapshots(shell: Shell, snapshots: [Snapshot]):
         names = [snapshot.name for snapshot in snapshots]
 
         logger.info("On %r for dataset %r destroying snapshots %r", shell, dataset, names)
-        args = ["zfs", "destroy", f"{dataset}@" + ",".join(names)]
 
-        shell.exec(args)
+        while names:
+            args = ["zfs", "destroy", f"{dataset}@" + ",".join(names)]
+            try:
+                shell.exec(args)
+                return
+            except ExecException as e:
+                m = re.search(r"cannot destroy snapshot .+?@(.+?): dataset is busy", e.stdout)
+                if m is None:
+                    raise
+
+                name = m.group(1)
+                logger.info("Snapshot %r on dataset %r is being hold, skipping", name, dataset)
```

Comment:
Hmm, can you guarantee that just by being busy? It could be cloned, or mounted (cwd in the .zfs dir?)

Reply:
Well, we can't do anything anyway, or are you just proposing to change log message?

Reply:
Sorry yes, just the log message, saying "hold" may confuse people with snapshos on hold (man zfs-hold)